Doc/Makefile

```diff
@@ -12,7 +12,7 @@ PAPER        =
 SOURCES      =
 DISTVERSION  = $(shell $(PYTHON) tools/extensions/patchlevel.py)
 
-ALLSPHINXOPTS = -b $(BUILDER) -d build/doctrees -W -D latex_elements.papersize=$(PAPER) \
+ALLSPHINXOPTS = -b $(BUILDER) -d build/doctrees -D latex_elements.papersize=$(PAPER) \
```

Comment:
I am fine with this change, but for the dev, could we have a specific target where we activate the '-W' flag?
my workflow
```shell
./configure --prefix=$PWD-build --with-pydebug --silent
make -j 4 -s 
make PYTHON=../python -C Doc/ venv
make -C Doc/ html 
``` 

but with `make html` if there is no check, we could commit an issue... and this error will appear once on the CI, and we have to push a new fix :-1: